android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncPerformanceTests.kt

```diff
@@ -0,0 +1,536 @@
+package com.mongodb.stitch.android.services.mongodb.remote.internal
+
+import android.os.Environment
+import android.os.StatFs
+import android.support.test.InstrumentationRegistry
+import android.util.Log
+
+import com.google.android.gms.tasks.Tasks
+
+import com.mongodb.MongoNamespace
+import com.mongodb.stitch.android.core.Stitch
+import com.mongodb.stitch.android.core.StitchAppClient
+import com.mongodb.stitch.android.services.mongodb.local.internal.AndroidEmbeddedMongoClientFactory
+import com.mongodb.stitch.android.services.mongodb.remote.RemoteMongoClient
+import com.mongodb.stitch.android.services.mongodb.remote.RemoteMongoCollection
+import com.mongodb.stitch.android.testutils.BaseStitchAndroidIntTest
+import com.mongodb.stitch.core.admin.Apps
+import com.mongodb.stitch.core.admin.authProviders.ProviderConfigs
+import com.mongodb.stitch.core.admin.services.ServiceConfigs
+import com.mongodb.stitch.core.admin.services.rules.RuleCreator
+import com.mongodb.stitch.core.admin.services.rules.RuleResponse
+import com.mongodb.stitch.core.auth.providers.anonymous.AnonymousCredential
+import com.mongodb.stitch.core.services.mongodb.remote.sync.internal.DataSynchronizer
+import com.mongodb.stitch.core.testutils.BaseStitchIntTest
+
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlin.system.measureTimeMillis
+
+import org.bson.BsonArray
+import org.bson.BsonDateTime
+import org.bson.BsonDouble
+import org.bson.BsonInt32
+import org.bson.BsonInt64
+import org.bson.BsonString
+import org.bson.types.Binary
+import org.bson.Document
+import org.bson.types.ObjectId
+
+import org.junit.Assert.assertEquals
+import org.junit.After
+import org.junit.Assume
+import org.junit.Before
+import org.junit.Test
+
+import java.util.Date
+
+class SyncPerformanceTests : BaseStitchAndroidIntTest() {
```

Comment:
[minor] as we spoke about offline, the tests should be broken out from the harness